### DISCUSSION TITLE
`gleam add` should try to keep `[dependencies]` sorted

### DISCUSSION THREAD

**Discussion by bonsairobo** [gleam-lang/gleam#3174] [Ideas & suggestions]
I find it useful to have my `[dependencies]` sorted. Much easier to read. Not sure about other folks, but the current behavior of appending to the end of the list seems arbitrary.

# Option A

`gleam add` assumes `[dependencies]` is already sorted and binary searches for an insertion position.

# Option B

Always sort `[dependencies]` after inserting a new element. This might annoy people if they have a particular organization strategy, although I can't think of any examples.

If people like to do something like:

```toml
[dependencies]

# One group of packages
foo = ...
bar = ...

# Another group
baz = ...
```

then `gleam add` could sort the groups separately. This might be annoying to implement though, assuming the first stage of updating `gleam.toml` is to parse the TOML. I'm not sure if this code makes any attempt to preserve spacing, etc.

# Option C

Check if the list is already sorted. If not, `gleam add` falls back to just inserting at the end.

# Option D

Add a `--sort` flag to `gleam add`. Or invert the default to be `--unsorted`. Depends on what we want the default behavior to be.

**Comment by inoas** [comment]
My wish would be automatically sorted but comments create new groups with a new line above unless it is directly below the section tag

**Reply by jrstrunk** [reply]
Comments or a newline would have been manually done by a user, so it's probably safe to assume organization blocks are separated by either.
